### PR TITLE
add range filters to filter on dates and other ranges

### PR DESCRIPTION
**Pull request by helaot**

### CODE REVIEW

File: commons/python/business_solutions_api/dataiku_formula.py [hunk 1]
```diff
@@ -9,18 +10,33 @@ def __init__(self):
...
+        expressions = []
+        expression = ''
+        if isinstance(vals, list):
+            sanitized_column = _escape_double_quotes(column)
+            sanitized_vals = [f'"{_escape_double_quotes(val)}"' for val in vals]
+            expression = f'arrayContains([{", ".join(sanitized_vals)}], strval("{sanitized_column}"))'
+        elif isinstance(vals, dict):
+            from_val = vals.get('from')
+            to_val = vals.get('to')
+            type = vals.get('type')
```
Comment (levkz) [inline commons/python/business_solutions_api/dataiku_formula.py:24]: type is built-in python class

File: commons/python/business_solutions_api/dataiku_formula.py [hunk 2]
```diff
@@ -9,18 +10,33 @@ def __init__(self):
         self._expressions = []
 
     @staticmethod
-    def _create_filter_by_column_vals(column: "str", vals: "list[str]"):
+    def _create_filter_by_column_vals(column: str, vals: Union["list[str]", dict]):
```
Comment (levkz) [inline commons/python/business_solutions_api/dataiku_formula.py:13]: I would create two sepparate functions - one `_create_filter_by_column_vals` and another `_create_filter_by_column_range` and choosing which one to use shouldn't be part of dataikuFormula class logic I think

**Review by levkz — APPROVED** [review]
LGTM!

A few small nitpick as always, good job!

File: commons/python/business_solutions_api/dataiku_formula.py (hunk 2 above)
Comment (helaot) [reply commons/python/business_solutions_api/dataiku_formula.py:13]: it is a possibility but I don't think there is a right or wrong way of doing it as both have their pros and cons

**Comment by helaot** [comment]
The new Range filters have been added to Aggrid PR : https://github.com/dataiku/solutions-contrib/pull/65. No need for this PR anymore

**Pull request closed without merging**